doc/source/performance/jemalloc-profiling.rst

```diff
@@ -0,0 +1,234 @@
+.. _jemalloc-profiling:
+
+=================================================
+Jemalloc Memory Allocation Profiling
+=================================================
+
+Implemented in :rn:`8.0.25-15`, |Percona Server| can take advantage of the memory-profiling ability of the jemalloc allocator. This ability provides a method to investigate memory-related issues.
+
+
+Requirements
+----------------
+
+This memory-profiling requires :ref:`jemalloc_detected`. This read-only variable returns ``true`` if jemalloc with the profiling-enabled option is being used by |Percona Server|. 
+
+As root, customize jemalloc with the following flags:
+
+.. list-table::
+    :widths: 20 40
+    :header-rows: 1
+
+    * - Option
+      - Description
+    * - `--enable-stats`
+      - Enables statistics-gathering ability
+    * - `--enable-prof`
+      - Enables heap profiling and the ability to detect leaks. 
+
+Using ``LD_PRELOAD``. Build the library, configure the malloc configuration with the ``prof:true`` string, and then use ``LD_PRELOAD`` to  preload the ``libjemalloc.so`` library. The libprocess ``MemoryProfiler`` class detects the library automatically and enables the profiling support.
+
+The following is an example of the required commands:
+
+.. sourcecode:: bash
+
+    ./configure --enable-stats --enable-prof && make && make install
+    MALLOC_CONF=prof:true 
+    LD_PRELOAD=/usr/lib/libjemalloc.so
+
+.. note:
+
+    Ensure the ``libjemalloc.so`` exists in the LD path.
+
+Use |Percona Server| with jemalloc with profiling enabled
+----------------------------------------------------------
+
+To detect if jemalloc is set, run the following command:
+
+.. sourcecode::  bash
+
+    jemalloc_detected ON
```

Comment:
I think this should be `SELECT @@jemalloc_detected;`?

Reply:
Fixed.